Common/cpp/Tools/BackgroundQueue.cpp

```diff
@@ -0,0 +1,97 @@
+#include "BackgroundQueue.h"
+
+#include <utility>
+
+namespace reanimated {
+
+BackgroundQueue::BackgroundQueue(const std::string &name) : name_(name) {
+  thread_ = std::thread([this] {
+#if __APPLE__
+    pthread_setname_np(name_.c_str());
+#endif
+    runLoop();
+  });
+#ifdef ANDROID
+  pthread_setname_np(thread_.native_handle(), name_.c_str());
+#endif
+}
+
+BackgroundQueue::~BackgroundQueue() {
+  running_ = false;
+  cv_.notify_all();
+  thread_.join();
+}
+
+jsi::Value BackgroundQueue::get(
+    jsi::Runtime &rt,
+    const jsi::PropNameID &propName) {
+  auto name = propName.utf8(rt);
+  if (name == "toString") {
+    return jsi::Function::createFromHostFunction(
+        rt,
+        propName,
+        0,
+        [this](jsi::Runtime &rt, const jsi::Value &, const jsi::Value *, size_t)
+            -> jsi::Value {
+          return jsi::String::createFromUtf8(rt, toString());
+        });
+  }
+  if (name == "name") {
+    return jsi::String::createFromUtf8(rt, name_);
+  }
+  return jsi::Value::undefined();
```

Comment:
Do we actually want to return undefined here instead of raising some errors? Or is it just a pattern for HostObjects?

Reply:
In pure JS, if you have `const foo = { bar: 1 };` then accessing `foo.xyz` returns `undefined`. So I'd say this is a pattern for JS objects and here I'm just trying to mimic it on a `jsi::HostObject`. And yes, this is also how other people implement `jsi::HostObject::get`.